tests/Unit/Domain.LinnApps.Tests/PurchaseOrderDeliveryServiceTests/WhenUpdatingShouldDecreaseOrderLineQty.cs

```diff
@@ -0,0 +1,132 @@
+                                                 }
+                                         };
+
+            // Order will be updated to delivery total.
+            this.line = new PurchaseOrderDetail
+            {
+                Line = 1,
+                OrderUnitPriceCurrency = 666,
+                OrderConversionFactor = 1.5m,
+                BaseOurUnitPrice = 111,
+                PurchaseDeliveries = new List<PurchaseOrderDelivery>(),
+                OurQty = 100,
+                OrderQty = 100
+            };
+            this.order = new PurchaseOrder
+            {
+                OrderNumber = 123456,
+                Details = new List<PurchaseOrderDetail> { this.line },
+                SupplierId = 9876543,
+                DocumentTypeName = "PO"
+            };
+            this.PurchaseOrderRepository.FindById(this.order.OrderNumber).Returns(this.order);
+
+            this.PurchaseOrdersPack.GetVatAmountSupplier(
+                this.line.BaseOurUnitPrice.GetValueOrDefault() * this.updatedDeliveries.First().OurDeliveryQty.GetValueOrDefault(),
+                this.order.SupplierId).Returns(this.baseVatAmount);
+
+            this.PurchaseOrdersPack.GetVatAmountSupplier(
+                this.line.OrderUnitPriceCurrency.GetValueOrDefault()
+                * this.updatedDeliveries.First().OurDeliveryQty.GetValueOrDefault(),
+                this.order.SupplierId).Returns(this.vatAmount);
+
+            this.AuthService.HasPermissionFor(
+                    AuthorisedAction.PurchaseOrderUpdate, Arg.Any<IEnumerable<string>>())
+                .Returns(true);
+
+            this.MiniOrderRepository.FindById(this.order.OrderNumber).Returns(new MiniOrder { });
+            this.result = this.Sut.UpdateDeliveriesForOrderLine(
+                this.order.OrderNumber,
+                this.line.Line,
+                this.updatedDeliveries,
+                new List<string>());
+        }
+
+        [Test]
+        public void ShouldReturnResult()
+        {
+            //Delivery fields
+            this.result.Count().Should().Be(1);
+            var updateData = this.updatedDeliveries.First();
+            var updated = this.result.First();
+            updated.DeliverySeq.Should().Be(1);
+            updated.OurDeliveryQty.Should().Be(updateData.OurDeliveryQty);
+            updated.OrderDeliveryQty.Should().Be(updateData.OurDeliveryQty / this.line.OrderConversionFactor);
+            updated.NetTotalCurrency.Should().Be(
+                Math.Round(
+                    updateData.OurDeliveryQty.GetValueOrDefault() * this.line.OurUnitPriceCurrency.GetValueOrDefault(),
+                    2));
+            updated.VatTotalCurrency.Should().Be(this.vatAmount);
+            updated.DeliveryTotalCurrency.Should().Be(
+                Math.Round(
+                    this.line.OrderUnitPriceCurrency.GetValueOrDefault() * updated.OurDeliveryQty.GetValueOrDefault(),
```

Comment:
I think this is ok actually on second thoughts after reading your other comments. Since it so happens that there is only one delivery the sum all quantities is just that first quantity and everything checks out ok